op-signer/gen-local-creds.sh

```diff
@@ -0,0 +1,189 @@
+USE_DOCKER=${OP_SIGNER_GEN_TLS_DOCKER:-true}
+
+# Helper function to run openssl commands
+run_openssl() {
+    if [ "$USE_DOCKER" = "true" ]; then
+        docker run --rm \
+            -v "$TLS_DIR:$TLS_DIR" \
+            -u "$USER_UID:$USER_GID" \
+            "$OPENSSL_IMAGE" "$@"
+    else
+        # Check if openssl is available locally
+        if ! command -v openssl &> /dev/null; then
+            echo "Error: OpenSSL is not installed locally. Please install OpenSSL or use Docker by setting OP_SIGNER_GEN_TLS_DOCKER=true"
+            exit 1
+        fi
+        openssl "$@"
+    fi
+}
+
+generate_ca() {
+    local force="$1"
+    [ "$force" = "true" ] || [ ! -f "$CA_CERT" ] || return 0
+
+    echo
+    echo "Generating CA..."
+
+    run_openssl req -newkey "rsa:$MOD_LENGTH" \
+        -new -nodes -x509 \
+        -days 365 \
+        -sha256 \
+        -out "$CA_CERT" \
+        -keyout "$CA_KEY" \
+        -subj "/O=$CERT_ORG_NAME/CN=root"
+}
+
+generate_client_tls() {
+    local hostname="$1"
+    echo
+    echo "Generating client TLS credentials for $hostname..."
+    
+    
+    # Create a directory for this client's credentials
+    local clientDir="$TLS_DIR/$hostname"
+    mkdir -p "$clientDir"
+    
+    # Generate client key
+    echo "Generating client key..."
+    run_openssl genrsa -out "$clientDir/$CLIENT_TLS_KEY" "$MOD_LENGTH"
+
+    local confFile="$clientDir/$CLIENT_OPENSSL_CNF"
+    
+    # Create a config file for the CSR
+    cat > "$confFile" << EOF
+[req]
+distinguished_name=req
+[san]
+subjectAltName=DNS:$hostname
+EOF
+    
+    echo "Generating client certificate signing request..."
+    run_openssl req -new -key "$clientDir/$CLIENT_TLS_KEY" \
+        -sha256 \
+        -out "$clientDir/$CLIENT_TLS_CSR" \
+        -subj "/O=$CERT_ORG_NAME/CN=$hostname" \
+        -extensions san \
+        -config "$confFile"
+    
+    echo "Generating client certificate..."
+    run_openssl x509 -req -in "$clientDir/$CLIENT_TLS_CSR" \
+        -sha256 \
+        -CA "$CA_CERT" \
+        -CAkey "$CA_KEY" \
+        -CAcreateserial \
+        -out "$clientDir/$CLIENT_TLS_CERT" \
+        -days 3 \
+        -extensions san \
+        -extfile "$confFile"
+}
+
+generate_client_key() {
+    local hostname="$1"
+    echo
+    echo "Generating private key for $hostname..."
+    local clientDir="$TLS_DIR/$hostname"
+    mkdir -p "$clientDir"
+    run_openssl ecparam -name secp256k1 -genkey -noout -param_enc explicit \
+        -out "$clientDir/$CLIENT_PRIVATE_KEY"
+}
+
+
+generate_client_credentials() {
+    setup_client_hostnames "$@"
+    process_clients generate_client_tls
+    process_clients generate_client_key
+}
+
+setup_client_hostnames() {
+    CLIENT_HOSTNAMES="$*"
+    if [ -z "$CLIENT_HOSTNAMES" ]; then
+        CLIENT_HOSTNAMES="localhost"
+    fi
+    printf "\nProcessing clients: %s\n" "$CLIENT_HOSTNAMES"
+}
+
+process_clients() {
+    generator="$1"
+    for hostname in $CLIENT_HOSTNAMES; do
```

Comment:
It looks like this will generate a certificate per hostname in the case of `process_clients  generate_client_tls`. Why create a cert per hostname instead of including multiple DNS SAN in the single cert? I think multiple SAN is the real world way a client would be identified by multiple DNS names

Reply:
the intention is to create multiple keys to be used by multiple services (see [here](https://github.com/ethpandaops/optimism-package/pull/207/files#diff-8952ca36c0cbca2168c99c185e468d02357daaab464a93cd7bf58bf8f3b90eb0R338))